src/runtime/analytics-service-test.js

```diff
@@ -287,6 +294,26 @@ describes.realWin('AnalyticsService', {}, env => {
       expect(loggedErrors.length).to.equal(1);
       expect(loggedErrors[0]).to.equal('Error when logging: ' + err);
     });
+
+    it('Should work without waiting for the promise', async () => {
+      // This sends another event and waits for it to be sent
+      eventManagerCallback({
+        eventType: AnalyticsEvent.IMPRESSION_PAYWALL,
+        eventOriginator: EventOriginator.SWG_CLIENT,
+        isFromUserAction: true,
+        additionalParameters: {droppedData: true},
+      });
+      await analyticsService.lastAction_;
```

Comment:
Does it make sense to verify that that lastAction_ is not null before waiting for it (apparently waiting for non-promises is valid and returns the non-promise)?